src/MongoDB.Driver/PipelineDefinition.cs

```diff
@@ -350,10 +350,8 @@ private static List<IPipelineStageDefinition> VerifyStages(List<IPipelineStageDe
 
             if (nextInputType != typeof(TOutput))
             {
-                var message = string.Format(
-                    "The output type to the last stage was expected to be {0}, but was {1}.",
-                    nextInputType,
-                    stages.Last().OutputType);
+                var message =
+                    $"The output type to the last stage was expected to be {nextInputType}, but was {typeof(TOutput)}.";
```

Comment:
The goal was to mirror the message being created on lines 340-344.

Of course this is better. Do you now want to change lines 340-344 to use string interpolation also?

Reply:
The error message is backward. 

The output of the last stage is expected to be `typeof(TOutput)`

Reply:
nit: use `nameof(stages)`

Reply:
Summary:

    var message = $"The output type to the last stage was expected to be {typeof(TOutput)}, but was {nextInputType}.";
    throw new ArgumentException(message, nameof(stages));